pkg/kopia/cmd/commands.go

```diff
@@ -0,0 +1,486 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"strconv"
+
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/kanisterio/kanister/pkg/logsafe"
+	"github.com/kanisterio/kanister/pkg/utils"
+)
+
+func bashCommand(args logsafe.Cmd) []string {
+	log.Info().Print("Kopia Command", field.M{"Command": args.String()})
+	return []string{"bash", "-o", "errexit", "-c", args.PlainText()}
+}
+
+func stringSliceCommand(args logsafe.Cmd) []string {
+	log.Info().Print("Kopia Command", field.M{"Command": args.String()})
+	return args.StringSliceCMD()
+}
+
+func kopiaArgs(password, configFilePath, logDirectory string, requireInfoLevel bool) logsafe.Cmd {
+	c := logsafe.NewLoggable(kopiaCommand)
+	if requireInfoLevel {
+		c = c.AppendLoggable(logLevelInfoFlag)
+	} else {
+		c = c.AppendLoggable(logLevelErrorFlag)
+	}
+	if configFilePath != "" {
+		c = c.AppendLoggableKV(configFileFlag, configFilePath)
+	}
+	if logDirectory != "" {
+		c = c.AppendLoggableKV(logDirectoryFlag, logDirectory)
+	}
+	if password != "" {
+		c = c.AppendRedactedKV(passwordFlag, password)
+	}
+	return c
+}
+
+// ExecKopiaArgs returns the basic Argv for executing kopia with the given config file path.
+func ExecKopiaArgs(configFilePath string) []string {
+	return kopiaArgs("", configFilePath, "", false).StringSliceCMD()
+}
+
```

Comment:
Does it make sense to add a struct CommandArgs:
```
type CommandArgs struct {
  encryptionKey string
  configFilePath string
  ...
}
```
Instead of having to pass the ordered arguments for each command wrapper, we can instead pass the struct. We can have specific arguments that are additionally needed for specific commands or extend this struct in other structs.

Command wrappers could also be simplified that way.

Reply:
I like the suggestion too 👍🏼

Reply:
Added CommandArgs in commits;
- https://github.com/kanisterio/kanister/pull/1552/commits/314f509377f63ea1c045a995ff7f8f3160300111
- https://github.com/kanisterio/kanister/pull/1557/commits/9783a391ab52243d65e48bb34b65d979ff92c5ee